File: airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination/buffered_stream_consumer/RecordWriter.java
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.cdk.integrations.destination.buffered_stream_consumer;

import io.airbyte.commons.functional.CheckedBiConsumer;
import io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair;
import java.util.stream.Stream;

public interface RecordWriter<T> extends CheckedBiConsumer<AirbyteStreamNameNamespacePair, Stream<T>, Exception> {

  @Override
  void accept(AirbyteStreamNameNamespacePair stream, Stream<T> records) throws Exception;

}

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/destination/buffered_stream_consumer/RecordWriter.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.integrations.destination.buffered_stream_consumer

import io.airbyte.commons.functional.CheckedBiConsumer
import io.airbyte.protocol.models.v0.AirbyteStreamNameNamespacePair

fun interface RecordWriter<T> :
    CheckedBiConsumer<AirbyteStreamNameNamespacePair, List<T>, Exception> {
    @Throws(Exception::class)
    override fun accept(stream: AirbyteStreamNameNamespacePair, records: List<T>)
}
